lib/src/sdk_env.dart

```diff
@@ -468,18 +468,36 @@ class ToolEnvironment {
     return DartdocResult(pr, pr.exitCode == 15, hasIndexHtml, hasIndexJson);
   }
 
-  /// Removes the dev_dependencies from the pubspec.yaml
+  /// Removes the `dev_dependencies` from the `pubspec.yaml`,
+  /// and also removes `pubspec_overrides.yaml`.
+  ///
   /// Adds lower-bound minimal SDK constraint - if missing.
-  /// Returns the backup file with the original content.
+  ///
+  /// Returns the result of the inner function, and restores the
+  /// original content upon return.
   Future<R> _withStripAndAugmentPubspecYaml<R>(
     String packageDir,
     FutureOr<R> Function() fn,
   ) async {
-    final backup = await _stripAndAugmentPubspecYaml(packageDir);
+    final now = DateTime.now();
+    final pubspecBackup = await _stripAndAugmentPubspecYaml(packageDir);
+
+    // backup of pubspec_overrides.yaml
+    final poyamlFile = File(p.join(packageDir, 'pubspec_overrides.yaml'));
+    File? poyamlBackup;
+    if (await poyamlFile.exists()) {
```

Comment:
Usually the "-sync" methods are both faster and more convenient
```suggestion
    if (poyamlFile.existsSync()) {
```

Reply:
Keeping it for consistency, and also we don't want to block the analysis VMs (yet, until sandboxing at least).